aws/src/main/java/org/apache/iceberg/aws/LakeFormationAwsClientFactory.java

```diff
@@ -0,0 +1,173 @@
+    if (properties.containsKey(AwsProperties.S3_USE_LAKE_FORMATION_CREDENTIALS) &&
+        properties.get(AwsProperties.S3_USE_LAKE_FORMATION_CREDENTIALS)
+            .equals(String.valueOf(true))) {
+      useLakeFormationCredentials = true;
+    }
+    this.region = region(properties);
+    this.s3Endpoint = properties.get(AwsProperties.S3FILEIO_ENDPOINT);
+    this.accessKeyId = properties.get(AwsProperties.S3FILEIO_ACCESS_KEY_ID);
+    this.secretAccessKey = properties.get(AwsProperties.S3FILEIO_SECRET_ACCESS_KEY);
+    this.sessionToken = properties.get(AwsProperties.S3FILEIO_SESSION_TOKEN);
+    this.roleArn = properties.get(AwsProperties.CLIENT_ASSUME_ROLE_ARN);
+    Preconditions.checkNotNull(roleArn, "Cannot initialize LakeFormationAwsClientFactory with empty roleArn");
+    this.timeout = PropertyUtil.propertyAsInt(properties,
+        AwsProperties.CLIENT_ASSUME_ROLE_TIMEOUT_SEC, AwsProperties.CLIENT_ASSUME_ROLE_TIMEOUT_SEC_DEFAULT);
+    this.externalId = properties.get(AwsProperties.CLIENT_ASSUME_ROLE_EXTERNAL_ID);
+    String tagString = properties.get(AwsProperties.CLIENT_ASSUME_ROLE_TAGS);
+    Preconditions.checkNotNull(tagString, "Cannot initialize LakeFormationAwsClientFactory with empty tags");
+    List<String> tokens = Splitter.on(',').splitToList(tagString);
+    String key = tokens.get(0).substring("Key=".length());
+    String value = tokens.get(1).substring("Value=".length());
+    this.tag = Tag.builder().key(key).value(value).build();
+  }
+
+  @Override
+  public S3Client s3() {
+    return S3Client.builder()
+        .endpointOverride(URI.create(s3Endpoint))
+        .credentialsProvider(buildS3CredentialProvider(useLakeFormationCredentials))
+        .region(region)
+        .httpClientBuilder(UrlConnectionHttpClient.builder())
+        .build();
+  }
+
+  @Override
+  public GlueClient glue() {
+    return GlueClient.builder()
+        .credentialsProvider(getAwsCredentialsProvider())
+        .region(region)
+        .httpClientBuilder(UrlConnectionHttpClient.builder())
+        .build();
+  }
+
+  @Override
+  public LakeFormationClient lakeFormation() {
+    return LakeFormationClient.builder()
+        .credentialsProvider(getAssumeRoleAwsCredentialsProvider())
+        .region(region)
+        .httpClientBuilder(UrlConnectionHttpClient.builder())
+        .build();
+  }
+
+  @Override
+  public KmsClient kms() {
+    return KmsClient.builder()
+        .credentialsProvider(getAwsCredentialsProvider())
+        .region(region)
+        .httpClientBuilder(UrlConnectionHttpClient.builder())
+        .build();
+  }
+
+  @Override
+  public DynamoDbClient dynamo() {
+    return DynamoDbClient.builder()
+        .credentialsProvider(getAwsCredentialsProvider())
+        .region(region)
+        .httpClientBuilder(UrlConnectionHttpClient.builder())
+        .build();
+  }
+
+  private AwsCredentialsProvider buildS3CredentialProvider(boolean shouldUseLakeFormationCredentials) {
```

Comment:
looks like this can be a one-liner and no need for a hleper? something like 

```java
shouldUseLakeFormationCredentials ? new LakeFormationCredentialsProvider(properties) : getAwsCredentialsProvider();
```

also `shouldUseLakeFormationCredentials` name is too long, prefer short names in Iceberg codebase, e.g. `useLakeformation`